test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -53,31 +53,33 @@ def setUp(self):
         TestFunctional.setUp(self)
 
     @tags('cray', 'smoke')
-    def test_cray_login_job(self):
+    def test_cray_no_queued_job(self):
         """
         Check output of apstat -rn on Cray with no queued job.
-        The compute node summary should show that there are no nodes in use,
-        held, or down.  Check the that the number of up nodes is the same as
-        the number of available nodes.
-
+        The compute node summary should show that there are no nodes in
+        use, held, or down.
+        Check that the number of nodes that are up and in batch mode
+        (State = 'UP  B') is the same as the number of Cray compute nodes
+        in pbsnodes.
         Check pbsnodes -av output on a Cray system with no queued job.
         pbsnodes output should show that nodes are free and resources are
         available.
-
-        Submit a simple sleep job that requests to run on a login node
-        and expect that job to go in running state on a login node.
-        Verify that the job runs to completion and check job output/error.
         """
         cu = CrayUtils()
-        self.assertEqual(cu.count_node_in_state('resv'), 0,
-                         "No compute node should be having ALPS reservation")
-        self.assertEqual(cu.count_node_in_state('use'), 0,
-                         "No compute node should be in use")
-        nodes_up_apstat = cu.count_node_in_state('up')
+        self.assertTrue(cu.count_node_in_state('resv') is 0,
+                        "No compute node should be having ALPS reservation")
+        self.assertTrue(cu.count_node_in_state('use') is 0,
+                        "No compute node should be in use")
+
+        nodes_up_b = cu.count_node_State('UP  B')
```

Comment:
Can you please write a comment about what "UP B" and "UP I " means.

Reply:
Done. "UP B" is also mentioned in the test's docstring.